bulk_grades/views.py

```diff
@@ -114,6 +111,29 @@ def post(self, request, course_id, *args, **kwargs):  # pylint: disable=unused-a
                      data.get('waiting', False))
         return HttpResponse(json.dumps(data), content_type='application/json')
 
+    def initialize_processor(self, request, course_id):
+        """
+        Initialize GradeCSVProcessor.
+        """
+        assignment_grade_max = request.GET.get('assignmentGradeMax')
+        assignment_grade_min = request.GET.get('assignmentGradeMin')
+        if self.operation_id:
+            if self.processor.course_id != course_id:
```

Comment:
Still not right. You need to load the operation from the operation_id. Check the lines you deleted.